### DISCUSSION TITLE
Issues with v.4.0 versus the Production v2.0

### DISCUSSION THREAD

**Discussion by jdalling** [DOI-BOR/PyForecast#100] [Q&A] [answered]
I am using the GP production v2.0 because v4.0 looks like it still has some bugs. 
Questions,
1. Other than using PCA to eliminate collinearity, are there other issues people are aware of? or processes that work well?  
2. How are others users selecting the set of equations they send to the density analysis? 
3. Will we be able to seamlessly load v2.0 forecast files into v4.0 at some point?
4. The documentation and wiki only exist for v2.0. 
5. Does the TSC have oversight over Reclamation software?

**Comment by tjrocha** [comment, marked as answer]
My responses to your questions follow.
1. Obviously lots of issues as we work through development and testing. Anecdotally, I can say that most users find v4 to be easier to use than the previous version. V4 is also what we will be using for continued development moving forward.
2. This is really going to be defined by the local use-case. Some users might want to select the top-x models, some might use just the top model, some users might pick ones with the most predictors, etc. The C-PN preferred method is on our forecast redevelopment report currently undergoing peer review (as of 10/25/2021).
3. Unfortunately, no. The current version is sufficiently different enough that making it backwards compatible with the previous version would take some effort.
4. Yes, once we get V4 to a good spot, we will be updating the Wiki.
5. No. TSC has no say in anything that we develop.

**Reply by jdalling** [reply]
Thank you Jon. My understanding is that there are reports being developed that make recommendations for using combined models versus single models, as well as using predictors with at least 20 years of data, to improve performance. I think those will be really helpful. 

For consistency and continuity, I am curious about the USACE's centralized software management procedures, but maybe that isn't a good fit for Reclamation.
